packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -130,21 +138,19 @@ export function createTokenIntrospectionMiddleware({
               : undefined
         }
       }
-      await next()
     } catch (err) {
-      if (bypassError && err instanceof HttpError) {
-        ctx.set('WWW-Authenticate', `GNAP as_uri=${config.authServerGrantUrl}`)
-        return await next()
+      if (!(err instanceof OpenPaymentsServerRouteError)) {
+        throw err
       }
 
-      if (err instanceof HttpError && err.status === 401) {
-        ctx.status = 401
-        ctx.message = err.message
-        ctx.set('WWW-Authenticate', `GNAP as_uri=${config.authServerGrantUrl}`)
-      } else {
+      ctx.set('WWW-Authenticate', `GNAP as_uri=${config.authServerGrantUrl}`)
+
+      if (!bypassError) {
         throw err
       }
     }
+
+    await next()
```

Comment:
no feedback, just a little retrospective: I wish I would have not added `bypassError` flag here and instead factored that use-case out into a separate function (kinda similar to `throwIfSignatureInvalid` and `authenticatedStatusMiddleware`). makes changes like this a lot harder to reason about. I added when implementing the public incoming payment.

Changes look fine though... getting public payment works as expected.